src/main.zig

```diff
@@ -272,6 +286,167 @@ pub const Player = struct {
     }
 };
 
+pub const Recorder = struct {
+    data: backends.BackendRecorder,
+
+    pub fn deinit(self: Recorder) void {
+        return switch (self.data) {
+            inline else => |b| b.deinit(),
+        };
+    }
+
+    pub const StartError = error{
+        CannotRecord,
+        OutOfMemory,
+        SystemResources,
+    };
+
+    pub fn start(self: Recorder) StartError!void {
+        return switch (self.data) {
+            inline else => |b| b.start(),
+        };
+    }
+
+    pub const RecordError = error{
+        CannotRecord,
+        OutOfMemory,
+    };
+
+    pub fn record(self: Recorder) RecordError!void {
+        return switch (self.data) {
+            inline else => |b| b.record(),
+        };
+    }
+
+    pub const PauseError = error{
+        CannotPause,
+        OutOfMemory,
+    };
+
+    pub fn pause(self: Recorder) PauseError!void {
+        return switch (self.data) {
+            inline else => |b| b.pause(),
+        };
+    }
+
+    pub fn paused(self: Recorder) bool {
+        return switch (self.data) {
+            inline else => |b| b.paused(),
+        };
+    }
+
+    pub const SetVolumeError = error{
+        CannotSetVolume,
+    };
+
+    // confidence interval (±) depends on the device
+    pub fn setVolume(self: Recorder, vol: f32) SetVolumeError!void {
```

Comment:
I am curious/doubtful that other APIs will support pausing, setting volume, etc. when recording. Will be interesting to see

Reply:
yeah im not sure too. but adding them wasn't a big deal anyway.